src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,164 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(value);
+        if (size == 0) {
+            head = tail = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddedIndex(index);
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
```

Comment:
Here we can immediately check that the index will be of different size (and in this case we will call the previous add(value)). Further, instead of complex if...else constructions (typical errors), we can use an iterator in the core method that will return us the node by index( getNodeByIndex) and assign it to a new variable, for example newNext . Also, in this case, it will be useful for us to create newPred, which will contain the field newNext.next. This way we can create a new node and assign its value as the previous field to newNext. Below I will give an example of what I mean, but try to implement the condition yourself (it is similar to the condition in the add(value) method)
```
 public void add(T value, int index) {
        ....
        checkRangeForIndex(index);
        Node<T> newNext = getElement(index);
        Node<T> newPred = newNext.prev;
        Node<T> newNode = new Node<>(value, newPred, newNext);
        newNext.prev = newNode;
        if (newPred == null) {
            .....
        } else {
           .....
        }
        size++;
    }
```